lib/house.rb

```diff
@@ -0,0 +1,54 @@
+class House
+
+  attr_reader :address, :price, :rooms
+
+  def initialize(price, address)
+    @price = price[1..-1].to_i
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    rooms << room
+  end
+
+  def above_market_average?
+    price > 500000
+  end
+
+  def rooms_from_category(category)
+    rooms.find_all { |room| room.category == category }
+  end
+
+  def area
+    total_area = []
+    rooms.each { |room| total_area << room.area }
+    total_area.sum
+  end
+
+  def details
+    Hash.new
+    {"price" => price, "address" => address}
+  end
+
+  def price_per_square_foot
+    (price.to_f / area.to_f).round(2)
+  end
+
+  def rooms_sorted_by_area
+    own_area = {}
+    rooms.each do |room|
+      own_area[room] = room.area
+    end
+    own_area.keys.sort_by { |value| own_area[value] }.reverse
+  end
+
+  #I keep getting basement first?
+  def rooms_by_category
+    sorted_rooms = {}
+    rooms.each do |room|
+      sorted_rooms[room.category] = [room]
```

Comment:
you're setting the room category as the key equal to an array always with one element inside. So if there are multiple rooms for that category, you're always going to get the last one, because that array can only ever have one room, based on the way you wrote it. 

You could either check out `group_by`, which is probably how I would have done it (http://gregpark.io/blog/ruby-group-by/)

Or if you do it this way, you could have done something like this:

```
def rooms_by_category
    sorted_rooms = {}
    rooms.each do |room|
      sorted_rooms[room.category] = rooms_from_category(room.category)
    end
end
```
This isn't as efficient as group_by, but I'm fairly certain it would work!